bioptim/optimization/optimal_control_program.py

```diff
@@ -755,35 +755,34 @@ def add_param_constraint_to_str(param_dict: ParameterList):
             constraint_str += f"<br/>"
             return constraint_str
 
-        def add_parameters_to_str(list_constraints: list, string: str):
+        def add_parameters_to_str(list_constraints: Union[list, ParameterList], string: str):
             for param in list_constraints:
                 string += f"{param}: " \
-                                   f"{list_constraints[f'{param}']}" \
+                                   f"{list_constraints[param]}" \
                                    f"<br/>"
             string += f"<br/>"
             return string
 
-        def lagrange_to_str(l_nodes: list, phase_idx: int):
+        def lagrange_to_str(objective_list: ObjectiveList):
             lagrange_str = ""
-            objective_idx = 0
-            for objective in l_nodes[phase_idx][0]['lagrange']:
-                if l_nodes[phase_idx][0]['sliced_target_lagrange'][objective_idx] is not None:
-                    if l_nodes[phase_idx][0]['quadratic_lagrange'][objective_idx] is True:
-                        lagrange_str += f"({objective} - " \
-                                        f"{l_nodes[phase_idx][0]['sliced_target_lagrange'][objective_idx]})" \
-                                        f"<sup>2</sup><br/>"
-                    else:
-                        lagrange_str += f"{objective} - " \
-                                        f"{l_nodes[phase_idx][0]['sliced_target_lagrange'][objective_idx]}<br/>"
-                else:
-                    if l_nodes[phase_idx][0]['quadratic_lagrange'][objective_idx] is True:
-                        lagrange_str += f"({objective})<sup>2</sup><br/>"
+            for objective in objective_list:
+                obj = objective[0]['objective']
+                if isinstance(obj.type, ObjectiveFcn.Lagrange):
+                    if obj.sliced_target is not None:
+                        if obj.quadratic is True:
```

Comment:
"is True" is always useless `if obj.quadratic` or `if obj.quadratic is not None` depending on the desired behavior (favour the first one when possible)

Reply:
Done